daemon/libnetwork/drivers/macvlan/macvlan_joinleave.go

```diff
@@ -76,7 +76,7 @@ func (d *driver) Join(ctx context.Context, nid, eid string, sboxKey string, jinf
 		}
 		// parse and correlate the endpoint v6 address with the available v6 subnets
 		if ep.addrv6 != nil && len(n.config.Ipv6Subnets) > 0 {
-			s := n.getSubnetforIPv6(ep.addrv6)
+			s := getSubnetForIP(ep.addr, n.config.Ipv6Subnets)
```

Comment:
This'll fix the tests ...

```suggestion
				s := getSubnetForIP(ep.addr6, n.config.Ipv6Subnets)
```

Reply:
Ah, thanks! I saw it failed and already was suspecting something like this ☺️ thanks for finding it; I'll push the fix later to have CI do its thing